src/jv.c

```diff
@@ -1862,6 +1862,68 @@ jv jv_object_iter_value(jv object, int iter) {
 /*
  * Memory management
  */
+jv jv_unshare(jv input){
+	switch(jv_get_kind(input)){
+		case JV_KIND_INVALID:
+			if(!jv_invalid_has_msg(jv_copy(input))){
+				jv_free(input);
+				return jv_invalid();
+			}
+			return jv_invalid_with_msg(jv_unshare(jv_invalid_get_msg(jv_copy(input))));
+		case JV_KIND_OBJECT:
+		case JV_KIND_ARRAY:
+			{
+				jv keys = jv_keys(jv_copy(input));
+				size_t keys_length = jv_array_length(jv_copy(keys));
+				
+				jv output_object;
+				if(jv_get_kind(input) == JV_KIND_OBJECT){
+					output_object = jv_object();
+				}else{
+					output_object = jv_array();
+				}
+				
+				for(size_t i = 0; i < keys_length; i++){
+					jv key = JV_ARRAY(jv_unshare(jv_array_get(jv_copy(keys), i)));
+
+					output_object = jv_setpath(
+						output_object,key,
+						jv_unshare(
+							jv_getpath(jv_copy(output_object), jv_copy(key))
+						)
```

Comment:
Not sure how much faster using `jv_array_set`/`jv_object_set` would be, could skip building a path array at least?

Reply:
the main reason I did it the way I did it was to save on code, but yes doing it specific for arrays and objects would probably be faster

Reply:
then it would probably be better to separate between JV_KIND_OBJECT AND JV_KIND_ARRAY